docs/raw/serving-community-history.md

```diff
@@ -0,0 +1,406 @@
+
+```
+ 0 11 22 33 44 55 66 77 88 99
+10 11 12 13 14 15 16 17 18 19 
+```
+
+With a `pieceLength` of 10 bytes, A1 will fit into `20 / 10 = 2` pieces:
+
+```
+ 0 11 22 33 44 55 66 77 88 99 // piece[0] SHA1: 0x123
+10 11 12 13 14 15 16 17 18 19 // piece[1] SHA1: 0x456
+```
+
+#### **Example: With padding**
+
+Let `WakuMessageArchive` "A2" be of size 21 bytes:
+
+```
+ 0 11 22 33 44 55 66 77 88 99
+10 11 12 13 14 15 16 17 18 19
+20
+```
+
+With a `pieceLength` of 10 bytes, A2 will fit into `21 / 10 = 2` pieces. The remainder will introduce a third piece:
+
+```
+ 0 11 22 33 44 55 66 77 88 99 // piece[0] SHA1: 0x123
+10 11 12 13 14 15 16 17 18 19 // piece[1] SHA1: 0x456
+20                            // piece[2] SHA1: 0x789
+```
+
+The next `WakuMessageArchive` "A3" will be appended ("#3") to the existing data and occupy the remainding space of the third data piece. The piece at index 2 will now produce a different SHA1 hash:
+
+```
+ 0 11 22 33 44 55 66 77 88 99 // piece[0] SHA1: 0x123
+10 11 12 13 14 15 16 17 18 19 // piece[1] SHA1: 0x456
+20 #3 #3 #3 #3 #3 #3 #3 #3 #3 // piece[2] SHA1: 0xeef
+#3 #3 #3 #3 #3 #3 #3 #3 #3 #3 // piece[3]
+```
+
+By filling up the remainding space of the third piece with A2 using its `padding` field, it is guaranteed that its SHA1 will stay the same:
+
+```
+ 0 11 22 33 44 55 66 77 88 99 // piece[0] SHA1: 0x123
+10 11 12 13 14 15 16 17 18 19 // piece[1] SHA1: 0x456
+20  0  0  0  0  0  0  0  0  0 // piece[2] SHA1: 0x999
+#3 #3 #3 #3 #3 #3 #3 #3 #3 #3 // piece[3]
+#3 #3 #3 #3 #3 #3 #3 #3 #3 #3 // piece[4]
+```
+
+## Seeding message history archives
+
+The community owner node MUST seed the [generated torrent](#creating-message-archive-torrents) until a new message history archive is created.
+
+The community owner node SHOULD NOT seed torrents for older message history archives. Only one torrent at a time should be seeded.
+
+## Creating magnet links
+
+Once a torrent file for a message archive is created, the community owner node MUST derive a magnet link following the [Magnet URI scheme](https://en.wikipedia.org/wiki/Magnet_URI_scheme) using the underlying BitTorrent protocol client.
+
+## Message archive distribution
+
+Message archives are available via the BitTorrent network as they are being [seeded by the community owner node](#seeding-message-history-archives).
+Other community member nodes will download the message archives from the BitTorrent network once they receive a magnet link that contains a message archive index.
+
+The community owner node MUST send magnet links containing message archives and the message archive index to a special community channel. The topic of that special channel follows the following format:
+
+```
+/{application-name}/{version-of-the-application}/{content-topic-name}/{encoding}
+```
+
+All messages sent with this topic MUST be instances of `ApplicationMetadataMessage` ([6/PAYLOADS](/specs/6-payloads)) with a `payload` of `CommunityMessageArchiveIndex`.
```

Comment:
`CommunityMessageArchiveIndex` is going to be added? could not find it

Reply:
how about the type of message? I think by `CommunityMessageArchiveIndex` you meant it is the `type` of `ApplicationMetadataMessage`?

Reply:
What happened to the magnet link? shouldn't it be published in the channel?